broker/provider.go

```diff
@@ -23,4 +23,16 @@ type Provider interface {
 	Provision(ctx context.Context, instanceID string, params ProvisionParameters) error
 	Deprovision(ctx context.Context, instanceID string, params DeprovisionParameters) error
 	GetState(ctx context.Context, instanceID string) (ServiceState, string, error)
+	GenerateCredentials(ctx context.Context, instanceID, bindingID string) (*Credentials, error)
+	RevokeCredentials(ctx context.Context, instanceID, bindingID string) error
+}
+
+// Credentials are the connection parameters for Redis clients
+type Credentials struct {
+	Host       string `json:"host"`
+	Port       int64  `json:"port"`
```

Comment:
Is there a reason why `int64` is chosen here? The largest TCP port number should fit inside a `uint16`.

Reply:
Amazon returns it as int64 so we used int64 too.